File: math/7_ReverseInteger/Solution.cpp
class Solution {
public:
    int reverse(int x) {
        long long xReversed = 0;
        while(x) {
            int pop = x % 10;
            x /= 10;
            // cout << xReversed;
            if (xReversed > INT_MAX/10 || (xReversed == INT_MAX / 10 && pop > 7)) return 0;
            if (xReversed < INT_MIN/10 || (xReversed == INT_MIN / 10 && pop < -8)) return 0;
            xReversed = xReversed * 10 + pop;
        }
        return xReversed;
    }
};

File: math/7_ReverseInteger/Solution.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        sign = 1
        if x < 0:    
            sign = -1

        x = abs(x)
        x_rev = 0
        while x > 0:
            pop = x % 10
            x_rev = x_rev * 10 + pop
            x = x // 10

        # alternatively
        # compare positive number with 0x7fffffff
        # compare size x_rev.bit_length() > 31
        if x_rev > 2**31-1  or x_rev < -2**31:
            return 0

        return x_rev * sign